Add width and height content attributes to <model>

heycam opened the pull request:
#### cf600d68403f383e2755725ad982c2e82dc1bd27
<pre>
Add width and height content attributes to &lt;model&gt;
<a href="https://bugs.webkit.org/show_bug.cgi?id=243974">https://bugs.webkit.org/show_bug.cgi?id=243974</a>
rdar://99014926

Reviewed by Antoine Quint.

For consistency with &lt;img&gt; and &lt;video&gt;, &lt;model&gt; should have width and
height content attributes that set presentational style for the width
and height properties.

* LayoutTests/model-element/model-element-width-height-expected.txt: Added.
* LayoutTests/model-element/model-element-width-height-visual-expected.html: Added.
* LayoutTests/model-element/model-element-width-height-visual.html: Added.
* LayoutTests/model-element/model-element-width-height.html: Added.
* Source/WebCore/Modules/model-element/HTMLModelElement.cpp:
(WebCore::HTMLModelElement::collectPresentationalHintsForAttribute):
(WebCore::HTMLModelElement::hasPresentationalHintsForAttribute const):
* Source/WebCore/Modules/model-element/HTMLModelElement.h:
* Source/WebCore/Modules/model-element/HTMLModelElement.idl:

Canonical link: <a href="https://commits.webkit.org/257294@main">https://commits.webkit.org/257294@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0902ec42e96e15644ebe519222446fe2e163d77e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98475 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7684 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31608 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107904 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168174 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102421 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8200 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85076 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91024 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/104561 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104143 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/6222 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89776 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33214 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88040 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/21135 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/76155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1626 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/22664 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1548 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/45168 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5028 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6471 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42085 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2923 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

marcoscaceres reviewed (commented):
Lgtm... but wondering if we need a reftest to verify the size actually visually being set?

graouts commented on LayoutTests/model-element/model-element-width-height.html:
```diff
@@ -0,0 +1,66 @@
+<body>
+<script>
+test(() => {
+    let model = document.createElement("model");
```
Don't shy away from using `const` :) Or are you one of those C++ people who don't want to use it because it's not a _real_ `const`?

heycam replied:
I am somewhat anti-const because it's so much longer than "let"!

heycam committed:
Add width and height content attributes to <model>
https://bugs.webkit.org/show_bug.cgi?id=243974
rdar://99014926

Reviewed by Antoine Quint.

For consistency with <img> and <video>, <model> should have width and
height content attributes that set presentational style for the width
and height properties.

* LayoutTests/model-element/model-element-width-height-expected.txt: Added.
* LayoutTests/model-element/model-element-width-height-visual-expected.html: Added.
* LayoutTests/model-element/model-element-width-height-visual.html: Added.
* LayoutTests/model-element/model-element-width-height.html: Added.
* Source/WebCore/Modules/model-element/HTMLModelElement.cpp:
(WebCore::HTMLModelElement::collectPresentationalHintsForAttribute):
(WebCore::HTMLModelElement::hasPresentationalHintsForAttribute const):
* Source/WebCore/Modules/model-element/HTMLModelElement.h:
* Source/WebCore/Modules/model-element/HTMLModelElement.idl:

Canonical link: https://commits.webkit.org/257294@main
